skimage/restoration/deconvolution.py

```diff
@@ -390,3 +391,137 @@ def richardson_lucy(image, psf, iterations=50, clip=True):
+    in Fish et al., 1995. It is an iterative process where the PSF
+    and image is deconvolved, respectively.
+    It is more noise tolerant than other algorithms,
+    such as Ayers-Dainty and the Weiner filter algorithms
+    (taken from the paper).
+
+    The algorithm performs well with gaussian PSFs and can recover
+    them nicely without any prior knowledge. If one has already an
+    educated guess, one should pass the PSF as argument to the function.
+    Note, that the PSF should have the same shape as the image,
+    and the PSF should be centered.
+
+    Due to its nature, the algorithm may divide by 0.
+    The function catches this issue and aborts the iterative process.
+    Mostly, the optimal number of iterations is before this error may occur.
+
+    References
+    ----------
+    .. [1] Fish, D. A., A. M. Brinicombe, E. R. Pike, and J. G. Walker.
+           "Blind deconvolution by means of the Richardson–Lucy algorithm."
+           JOSA A 12, no. 1 (1995): 58-65.
+
+           https://pdfs.semanticscholar.org/9e3f/a71e22caf358dbe873e9649f08c205d0c0c0.pdf
+    """
+    if return_iterations:
+        all_iterations = np.empty((iterations, 2,) + image.shape)
+
+    # Convert image to float for computations
+    image = image.astype(np.float)
+
+    # Initialize im_deconv and PSF
+    im_deconv = np.full(image.shape, 0.5)
+
+    if psf is None:
+        psf = np.full(image.shape, 0.5)
+    else:
+        assert psf.shape == image.shape, \
+            'Image and PSF should have the same shape!'
+        psf = psf.astype(np.float)
+
+    for i in range(iterations):
+        # Deconvolve the PSF
+        # Hack: in original publication one would have used `image`,
+        #       however, this does not work.
+        #       Using `im_deconv` instead recovers PSF.
+        relative_blur_psf = im_deconv / fftconvolve(psf, im_deconv, 'same')
+
+        # Check if relative_blur_psf contains nan,
+        # causing the algorithm to fail
+        if np.count_nonzero(~np.isnan(relative_blur_psf)) \
+                < relative_blur_psf.size:
+            warnings.warn('Iterations stopped after {} iterations'
+                          ' because PSF contains zeros!'.format(i),
+                          RuntimeWarning)
+            break
+
+        else:
+            psf *= fftconvolve(relative_blur_psf,
+                               im_deconv[::-1, ::-1],
+                               'same')
+
+            # Compute inverse again
+            psf_mirror = psf[::-1, ::-1]
+
+            # Standard Richardson-Lucy deconvolution
+            relative_blur = image / fftconvolve(im_deconv, psf, 'same')
+            im_deconv *= fftconvolve(relative_blur, psf_mirror, 'same')
+
+            # Add iteration to list, if desired
+            if return_iterations:
+                all_iterations[i, 0] = im_deconv.copy()
+                all_iterations[i, 1] = psf.copy()
+
+    # Don't know if this makes sense here...
```

Comment:
Can you elaborate? See also my comment above. This is specific to float images...

Reply:
I am not sure about the clipping especially with PSFs that are derived internally, however, one can just disable the clip function and gets the desired result, maybe I just remove the comment.